### ISSUE TITLE
2nn Crowding Function Failing On Some Problems

### ISSUE THREAD

**Issue by jordan-cork** [mooscaliaproject/pymoode#8]
In file `pymoode\survival\rank_and_crowding\metrics.py`, `calc_2nn` (from `pymoode.cython.mnn`) is failing when 2nn is used as the crowding method in the NSDE algorithm. This occurs when the algorithm is run on RCM 27 (see reference at end of message). Notably, it is able to complete two generations before failing. Also, on all other RCM problems tried, it works.

Here is the traceback:
```
Traceback (most recent call last):
  File "C:\Users\jordan\PycharmProjects\algorithm_comparison\analysis.py", line 219, in <module>
    res = minimize(problem,
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\pymoo\optimize.py", line 67, in minimize
    res = algorithm.run()
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\pymoo\core\algorithm.py", line 138, in run
    self.next()
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\pymoo\core\algorithm.py", line 159, in next
    self.advance(infills=infills)
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\pymoo\core\algorithm.py", line 225, in advance
    val = self._advance(infills=infills, **kwargs)
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\pymoo\algorithms\base\genetic.py", line 109, in _advance
    self.pop = self.survival.do(self.problem, pop, n_survive=self.pop_size, algorithm=self, **kwargs)
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\pymoo\core\survival.py", line 45, in do
    survivors = self._do(problem, pop[feas], *args, n_survive=min(len(feas), n_survive), **kwargs)
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\pymoode\survival\rank_and_crowding\classes.py", line 110, in _do
    self.crowding_func.do(
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\pymoode\survival\rank_and_crowding\metrics.py", line 79, in do
    return self._do(F, n_remove=n_remove)
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\pymoode\survival\rank_and_crowding\metrics.py", line 129, in _do
    return super()._do(F, **kwargs)
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\pymoode\survival\rank_and_crowding\metrics.py", line 111, in _do
    _d = self.function(_F, **kwargs)
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\pymoode\survival\rank_and_crowding\metrics.py", line 31, in calc_2nn_nds
    return calc_2nn(F, n_remove=n_remove)
  File "pymoode\cython\mnn.pyx", line 94, in pymoode.cython.mnn.calc_2nn
  File "pymoode\cython\mnn.pyx", line 147, in pymoode.cython.mnn.c_calc_mnn
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\numpy\core\fromnumeric.py", line 858, in argpartition
    return _wrapfunc(a, 'argpartition', kth, axis=axis, kind=kind, order=order)
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\numpy\core\fromnumeric.py", line 56, in _wrapfunc
    return _wrapit(obj, method, *args, **kwds)
  File "C:\Users\jordan\anaconda3\envs\algorithm_footprint\lib\site-packages\numpy\core\fromnumeric.py", line 45, in _wrapit
    result = getattr(asarray(obj), method)(*args, **kwds)
ValueError: kth(=2) out of bounds (2)
```

RCM Problem Suite:
A. Kumar, G. Wu, M. Z. Ali, Q. Luo, R. Mallipeddi, P. N. Suganthan, S. Das, A Benchmark-Suite
of real-World constrained multi-objective optimization problems and some baseline results, Swarm
and Evolutionary Computation, Volume 67, 2021, 100961. doi:10.1016/j.swevo.2021.100961.

**Comment by bruscalia** [comment]
Hi, @jordan-cork! I believe you found an edge case in which the front has less than 3 elements. I'll take a look at this as soon as possible (which might take a little more than a week as I'm traveling on vacation right now). It's probably not going to be a long fix, so likely to include in 0.3.0. Many thanks for finding it!

**Comment by bruscalia** [comment]
Hi, @jordan-cork! I believe I fixed the problem. Can you try version 0.3.0.rc3? It was already uploaded into PyPI, so just hit `pip install pymoode==0.3.0.rc3`
